latex/chain.tex

```diff
@@ -0,0 +1,787 @@
+        \begin{array}{l}
+          \var{e}\\
+          \var{pp_{new}}\\
+          \var{blocks}\\
+        \end{array}
+      \right)
+        \vdash \var{es} \trans{epoch}{} \var{es}
+      }
+    }
+    {
+      \left(
+        {\begin{array}{c}
+           \var{bhbs} \\
+           \var{seeds} \\
+           \var{pp_{new}} \\
+         \end{array}}
+      \right)
+      \vdash
+      \left(
+        {\begin{array}{c}
+            \eta_0 \\
+            \var{blocks} \\
+            \var{es} \\
+         \end{array}}
+      \right)
+    \trans{newepoch}{\var{bhb}}
+      \left(
+        {\begin{array}{c}
+            \varUpdate{\var{seeds}\downarrow i} \\
+            \varUpdate{\emptyset} \\
+            \varUpdate{\var{es}'} \\
+         \end{array}}
+     \right)
+    }
+   \end{equation}
+
+   \nextdef
+
+  \begin{equation*}
+    \inference[Not-New-Epoch]
+    {
+      \var{last};\var{hs} \leteq bhbs
+      &
+      \var{e} \leteq \epoch{\bslot{bhb}}
+      &
+      \epoch{\bslot{last}} \geq e
+    }
+    {
+      \left(
+        {\begin{array}{c}
+           \var{bhbs} \\
+           \var{seeds} \\
+           \var{pp_{new}} \\
+         \end{array}}
+      \right)
+      \vdash
+      \left(
+        {\begin{array}{c}
+            \eta_0 \\
+            \var{blocks} \\
+            \var{es} \\
+         \end{array}}
+      \right)
+    \trans{newepoch}{\var{bhb}}
+      \left(
+        {\begin{array}{c}
+            \eta_0 \\
+            \var{blocks} \\
+            \var{es} \\
+         \end{array}}
+     \right)
+    }
+ \end{equation*}
+
+\caption{New Epoch rules}
+\label{fig:rules:new-epoch}
+\end{figure}
+
+\begin{note}
+  In order for $i$ above to be defined, there must be at least
+  $(2\cdot\mathsf{stableAfter} + 1)$-many blocks made in total.
+\end{note}
+
+\clearpage
+
+\subsection{Block Body Transition}
+\label{sec:block-trans}
+
+\begin{figure}[ht]
+  \emph{Block body environments}
+  \begin{equation*}
+    \BBodyEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \eta_0 & \Seed & \text{epoch nonce}\\
+        \var{pp} & \PParams & \text{protocol parameters}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Block body states}
+  \begin{equation*}
+    \BBodyState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{ls} & \LState & \text{ledger state}\\
+        \var{b} & \BlocksMade & \text{blocks made}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Block body transitions}
+  \begin{equation*}
+    \_ \vdash \var{\_} \trans{bbody}{\_} \var{\_} \subseteq
+    \powerset (\BBodyEnv \times \BBodyState \times \Block \times \BBodyState)
+  \end{equation*}
+  \caption{Block body transition-system types}
+  \label{fig:ts-types:block}
+\end{figure}
+
+\begin{figure}[ht]
+  \begin{equation}
+    \inference[Block-Body]
+    {
+      ((bhb,~\wcard),~\var{txs}) \leteq block
+      &
+      s \leteq \slotToSeed{\bslot{bhb}}
+      \\
+      \bseedn{bhb} = \eta_0\seedOp s\seedOp\Seedn
```

Comment:
do we require `seedOp` to be associative ?

Reply:
I don't think so, but it probably will be. We should either require it or at least say which way we associate. :)